packages/components/pie-modal/src/index.ts

```diff
@@ -31,21 +36,91 @@ export class PieModal extends RtlMixin(LitElement) {
         this.addEventListener('click', (event) => this._handleDialogLightDismiss(event));
     }
 
+    firstUpdated (changedProperties: DependentMap<ModalProps>) : void {
+        this._handleModalOpenStateOnFirstRender(changedProperties);
+    }
+
+    updated (changedProperties: DependentMap<ModalProps>) : void {
+        this._handleModalOpenStateChanged(changedProperties);
+    }
+
+    /**
+     * Opens the dialog element and disables page scrolling
+     */
+    private _handleModalOpened () : void {
+        disableBodyScroll(this);
+        //  We require this because toggling the prop `isOpen` itself won't
+        //  allow the dialog to open in the correct way (with the default background),
+        //  the method `showModal()` needs to be invoked.
+        this._dialog?.showModal();
+    }
+
+    /**
+     * Closes the dialog element and re-enables page scrolling
+     */
+    private _handleModalClosed () : void {
+        enableBodyScroll(this);
+        //  Closes the native dialog element
+        this._dialog?.close();
+    }
+
+    /**
+     * This is only to be used inside the component template as direct property
+     * reassignment is not allowed.
+     */
+    private _triggerCloseModal = () : void => {
+        this.isOpen = false;
+    };
+
+    connectedCallback () : void {
+        super.connectedCallback();
+        document.addEventListener(ON_MODAL_OPEN_EVENT, this._handleModalOpened.bind(this));
+        document.addEventListener(ON_MODAL_CLOSE_EVENT, this._handleModalClosed.bind(this));
+    }
+
+    disconnectedCallback () : void {
+        document.removeEventListener(ON_MODAL_OPEN_EVENT, this._handleModalOpened.bind(this));
+        document.removeEventListener(ON_MODAL_CLOSE_EVENT, this._handleModalClosed.bind(this));
+        super.disconnectedCallback();
+    }
+
+    // Handles the value of the isOpen property on first render of the component
+    private _handleModalOpenStateOnFirstRender (changedProperties: DependentMap<ModalProps>) : void {
+        // This ensures if the modal is open on first render, the scroll lock and backdrop are applied
+        const previousValue = changedProperties.get('isOpen');
+
+        if (previousValue === undefined && this.isOpen) {
+            this._dispatchModalOpenEvent();
+        }
+    }
+
+    // Handles changes to the modal isOpen property by dispatching any appropriate events
+    private _handleModalOpenStateChanged (changedProperties: DependentMap<ModalProps>) : void {
+        const previousValue = changedProperties.get('isOpen');
+
+        if (previousValue !== undefined) {
+            if (previousValue as boolean) {
```

Comment:
Is the `as boolean` necessary here? I would have thought the undefined check above should leave a boolean as the only other possibility.

Reply:
Logically yes, but not for intellisense. I'm happy to remove it though

Reply:
good spot - will remove